packages/editor-ui/src/components/ExecutionsUsage.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+	<div :class="$style.container">
+		<div v-if="!isTrialExpired() && trialHasExecutionsLeft()" :class="$style.usageText">
+			<i18n path="executionUsage.currentUsage">
+				<template #text>
+					<n8n-text size="xsmall" color="text-dark">
+						{{ locale.baseText('executionUsage.currentUsage.text') }}
+					</n8n-text>
+				</template>
+				<template #count>
+					<n8n-text size="xsmall" :bold="true" color="warning">
+						{{
+							locale.baseText('executionUsage.currentUsage.count', {
+								interpolate: { days: daysLeftOnTrial().toString() },
+							})
+						}}
+					</n8n-text>
+				</template>
+			</i18n>
+		</div>
+		<div v-if="isTrialExpired()" :class="$style.usageText">
+			<n8n-text size="xsmall" color="danger">
+				{{ locale.baseText('executionUsage.expired.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!trialHasExecutionsLeft()" :class="$style.usageText">
+			<n8n-text size="xsmall">
+				{{ locale.baseText('executionUsage.ranOutOfExecutions.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!isTrialExpired()" :class="$style.usageCounter">
+			<div>
+				<progress
+					:class="[
+						trialHasExecutionsLeft() ? $style.progressBarSuccess : $style.progressBarDanger,
+						$style.progressBar,
+					]"
+					:value="currentExecutions()"
+					:max="maxExecutions()"
+				></progress>
+			</div>
+			<div :class="$style.executionsCountSection">
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft() ? 'text-dark' : 'danger'">
+					{{ currentExecutions() }}/{{ maxExecutions() }}
+				</n8n-text>
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft() ? 'text-dark' : 'danger'">{{
+					locale.baseText('executionUsage.label.executions')
+				}}</n8n-text>
+			</div>
+		</div>
+
+		<div :class="$style.upgradeButtonSection">
+			<n8n-button
+				:label="locale.baseText('executionUsage.button.upgrade')"
+				size="mini"
+				icon="gem"
+				type="success"
+				:block="true"
+				@click="onUpgradeClicked"
+			/>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { i18n as locale } from '@/plugins/i18n';
+import { DateTime } from 'luxon';
+
+export interface CloudPlanData {
+	planId: number;
+	monthlyExecutionsLimit: number;
+	activeWorkflowsLimit: number;
+	credentialsLimit: number;
+	isActive: boolean;
+	displayName: string;
+	expirationDate: string;
+	metadata: PlanMetadata;
+	usage: Usage;
+}
+export interface PlanMetadata {
+	version: 'v1';
+	group: 'opt-out' | 'opt-in';
+	slug: 'pro-1' | 'pro-2' | 'starter' | 'trial-1';
+	trial?: Trial;
+}
+export interface Trial {
+	length: number;
+	gracePeriod: number;
+}
+
+export interface Usage {
+	executions: number;
+	activeWorkflows: number;
+}
+
+const props = defineProps<{ cloudPlanData: CloudPlanData }>();
```

Comment:
You could use the interface here without wrapping it in another object. Also, there's no need to duplicate the interfaces, only the root props interface needs to be exported from this file, the other ones could be imported. Take a look at an example in `ExecutionFilter.vue` component: 
```ts

import type {
	ExecutionFilterType,
	ExecutionFilterMetadata,
	IWorkflowShortResponse,
} from '@/Interface';

export type ExecutionFilterProps = {
	workflows?: IWorkflowShortResponse[];
	popoverPlacement?: PopoverPlacement;
};
const props = withDefaults(defineProps<ExecutionFilterProps>(), {
	popoverPlacement: 'bottom',
});
```